### PR TITLE
PC-5612 Add CloudWatch support.

### PR DESCRIPTION
**Pull request by davex98**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/nobl9/terraform-provider-nobl9
 go 1.16
 
 require (
-	github.com/google/uuid v1.1.2
+	github.com/google/uuid v1.3.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.7.0
-	github.com/nobl9/nobl9-go v0.0.0-20220506132101-a0e9c676ada0 // indirect
+	github.com/nobl9/nobl9-go v0.0.0-20220629130434-044750c773a8
```
Comment (kskitek) [inline go.mod:8]: I think we should tag github.com/nobl9/nobl9-go with something like v0.1.0 and use it here. My reasoning is that we want to actually start tracking changes in the SDK with n9 releases.
The same in `nobl9/n9` repo.

**Review by kskitek — COMMENTED** [review]
Do we want to add CloudWatch Agent support here as well, or do you plan to do that after we plan implementing missing features?

File: go.mod (hunk 1 above)
Comment (davex98) [reply go.mod:8]: I was thinking about creating a tag once we go 'public', but I am willing to create one that follows our n9 version - v1.33.0.

Comment (kskitek) [reply go.mod]: I just realized my message is not really an answer :D I am in favour of tagging it now with 1.33

Comment (davex98) [reply go.mod]: Great :D here it is! https://github.com/nobl9/terraform-provider-nobl9/pull/30/commits/7ddffba7c52d7db26f490334a26cf20b8d4e9983

**Comment by davex98** [comment]
```
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAcc_Nobl9Agent
--- SKIP: TestAcc_Nobl9Agent (0.00s)
=== RUN   TestAcc_Nobl9AlertPolicy
=== RUN   TestAcc_Nobl9AlertPolicy/alert-policy
=== RUN   TestAcc_Nobl9AlertPolicy/alert-policy-with-alert-method
=== RUN   TestAcc_Nobl9AlertPolicy/alert-policy-with-multi-alert-method
=== RUN   TestAcc_Nobl9AlertPolicy/alert-policy-with-multi-alert-method-reverse
--- PASS: TestAcc_Nobl9AlertPolicy (17.04s)
    --- PASS: TestAcc_Nobl9AlertPolicy/alert-policy (2.89s)
    --- PASS: TestAcc_Nobl9AlertPolicy/alert-policy-with-alert-method (4.75s)
    --- PASS: TestAcc_Nobl9AlertPolicy/alert-policy-with-multi-alert-method (5.10s)
    --- PASS: TestAcc_Nobl9AlertPolicy/alert-policy-with-multi-alert-method-reverse (4.30s)
=== RUN   TestAcc_Nobl9AlertMethod
=== RUN   TestAcc_Nobl9AlertMethod/test-webhook
=== PAUSE TestAcc_Nobl9AlertMethod/test-webhook
=== RUN   TestAcc_Nobl9AlertMethod/test-webhook-fields
=== PAUSE TestAcc_Nobl9AlertMethod/test-webhook-fields
=== RUN   TestAcc_Nobl9AlertMethod/test-pagerduty
=== PAUSE TestAcc_Nobl9AlertMethod/test-pagerduty
=== RUN   TestAcc_Nobl9AlertMethod/test-slack
=== PAUSE TestAcc_Nobl9AlertMethod/test-slack
=== RUN   TestAcc_Nobl9AlertMethod/test-discord
=== PAUSE TestAcc_Nobl9AlertMethod/test-discord
=== RUN   TestAcc_Nobl9AlertMethod/test-opsgenie
=== PAUSE TestAcc_Nobl9AlertMethod/test-opsgenie
=== RUN   TestAcc_Nobl9AlertMethod/test-servicenow
=== PAUSE TestAcc_Nobl9AlertMethod/test-servicenow
=== RUN   TestAcc_Nobl9AlertMethod/test-jira
=== PAUSE TestAcc_Nobl9AlertMethod/test-jira
=== RUN   TestAcc_Nobl9AlertMethod/test-teams
=== PAUSE TestAcc_Nobl9AlertMethod/test-teams
=== RUN   TestAcc_Nobl9AlertMethod/test-email
=== PAUSE TestAcc_Nobl9AlertMethod/test-email
=== CONT  TestAcc_Nobl9AlertMethod/test-webhook
=== CONT  TestAcc_Nobl9AlertMethod/test-opsgenie
=== CONT  TestAcc_Nobl9AlertMethod/test-slack
=== CONT  TestAcc_Nobl9AlertMethod/test-email
=== CONT  TestAcc_Nobl9AlertMethod/test-discord
=== CONT  TestAcc_Nobl9AlertMethod/test-jira
=== CONT  TestAcc_Nobl9AlertMethod/test-servicenow
=== CONT  TestAcc_Nobl9AlertMethod/test-teams
=== CONT  TestAcc_Nobl9AlertMethod/test-pagerduty
=== CONT  TestAcc_Nobl9AlertMethod/test-webhook-fields
--- PASS: TestAcc_Nobl9AlertMethod (0.00s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-webhook (2.49s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-opsgenie (2.55s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-slack (2.68s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-servicenow (2.74s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-teams (2.78s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-discord (2.79s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-jira (2.88s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-email (2.93s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-pagerduty (1.76s)
    --- PASS: TestAcc_Nobl9AlertMethod/test-webhook-fields (1.93s)
=== RUN   TestAcc_Nobl9Project
--- PASS: TestAcc_Nobl9Project (1.65s)
=== RUN   TestAcc_Nobl9RoleBinding
--- SKIP: TestAcc_Nobl9RoleBinding (0.00s)
=== RUN   TestAcc_Nobl9Service
--- PASS: TestAcc_Nobl9Service (1.54s)
=== RUN   TestAcc_Nobl9SLO
=== PAUSE TestAcc_Nobl9SLO
=== CONT  TestAcc_Nobl9SLO
=== RUN   TestAcc_Nobl9SLO/test-prometheus
=== RUN   TestAcc_Nobl9SLO/test-prom-with-ap
=== RUN   TestAcc_Nobl9SLO/test-prom-with-countmetrics
=== RUN   TestAcc_Nobl9SLO/test-prom-with-multiple-objectives
=== RUN   TestAcc_Nobl9SLO/test-prom-full
=== RUN   TestAcc_Nobl9SLO/test-prom-with-time-slices
=== RUN   TestAcc_Nobl9SLO/test-prom-with-raw-metric-in-objective
=== RUN   TestAcc_Nobl9SLO/test-prom-with-attachments
=== RUN   TestAcc_Nobl9SLO/test-newrelic
=== RUN   TestAcc_Nobl9SLO/test-appdynamics
=== RUN   TestAcc_Nobl9SLO/test-splunk
=== RUN   TestAcc_Nobl9SLO/test-lightstep
=== RUN   TestAcc_Nobl9SLO/test-splunk-observability
=== RUN   TestAcc_Nobl9SLO/test-dynatrace
=== RUN   TestAcc_Nobl9SLO/test-thousandeyes
=== RUN   TestAcc_Nobl9SLO/test-graphite
=== RUN   TestAcc_Nobl9SLO/test-bigquery
=== RUN   TestAcc_Nobl9SLO/test-opentsdb
=== RUN   TestAcc_Nobl9SLO/test-cloudwatch-with-stat
=== RUN   TestAcc_Nobl9SLO/test-cloudwatch-with-sql
=== RUN   TestAcc_Nobl9SLO/test-cloudwatch-with-json
=== RUN   TestAcc_Nobl9SLO/test-multiple-ap
=== RUN   TestAcc_Nobl9SLO/test-composite-occurrences
=== RUN   TestAcc_Nobl9SLO/test-composite-time-slices
--- PASS: TestAcc_Nobl9SLO (83.53s)
    --- PASS: TestAcc_Nobl9SLO/test-prometheus (3.56s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-ap (3.73s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-countmetrics (3.30s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-multiple-objectives (3.16s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-full (3.12s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-time-slices (3.42s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-raw-metric-in-objective (3.63s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-attachments (3.68s)
    --- PASS: TestAcc_Nobl9SLO/test-newrelic (3.47s)
    --- PASS: TestAcc_Nobl9SLO/test-appdynamics (3.08s)
    --- PASS: TestAcc_Nobl9SLO/test-splunk (3.29s)
    --- PASS: TestAcc_Nobl9SLO/test-lightstep (3.13s)
    --- PASS: TestAcc_Nobl9SLO/test-splunk-observability (3.31s)
    --- PASS: TestAcc_Nobl9SLO/test-dynatrace (3.46s)
    --- PASS: TestAcc_Nobl9SLO/test-thousandeyes (3.41s)
    --- PASS: TestAcc_Nobl9SLO/test-graphite (3.78s)
    --- PASS: TestAcc_Nobl9SLO/test-bigquery (3.44s)
    --- PASS: TestAcc_Nobl9SLO/test-opentsdb (3.95s)
    --- PASS: TestAcc_Nobl9SLO/test-cloudwatch-with-stat (3.37s)
    --- PASS: TestAcc_Nobl9SLO/test-cloudwatch-with-sql (3.45s)
    --- PASS: TestAcc_Nobl9SLO/test-cloudwatch-with-json (3.28s)
    --- PASS: TestAcc_Nobl9SLO/test-multiple-ap (4.66s)
    --- PASS: TestAcc_Nobl9SLO/test-composite-occurrences (3.37s)
    --- PASS: TestAcc_Nobl9SLO/test-composite-time-slices (3.48s)
PASS
ok  	github.com/nobl9/terraform-provider-nobl9/nobl9	(cached)

```

**Review by kskitek — APPROVED** [review]
lgtm!

Tests are working locally (failing on test organization due to missing agents).

**Pull request merged**